planner/core/rule_aggregation_elimination.go

```diff
@@ -64,6 +74,11 @@ func (*aggregationEliminateChecker) tryToEliminateAggregation(agg *LogicalAggreg
 		}
 	}
 	if coveredByUniqueKey {
+		if a.oldAggEliminationCheck {
+			if !CheckCanConvertAggToProj(agg) {
```

Comment:
And this check can always take effect, instead of only for the pushed agg?

Reply:
no.
agg1 -> join -> agg2
now the case is that agg1's work has been transferred to agg2, agg1 can be just removed.
while 'we' do the remove-work by aggEliminator, in which it will rewrite some agg func to a new projection, consequently causing problems.

the ideal approach is that to make the agg1's parent shift its args to what join can produce after agg2 is generated, otherwise, the can't-find-column error will occur. (currently too complicated for a quick fix)

we just keep the old agg1 remained, but its work is actually done by agg2, old-agg1 here acts like a role to project what schema the join produced to what the old-agg1 parent wanted as before.